x/model/keeper/msg_server_model.go

```diff
@@ -125,6 +127,15 @@ func (k msgServer) UpdateModel(goCtx context.Context, msg *types.MsgUpdateModel)
 		model.ProductUrl = msg.ProductUrl
 	}
 
+	if msg.LsfUrl != "" && msg.LsfUrl != model.LsfUrl {
```

Comment:
What if `msg.LsfUrl == model.LsfUrl` and `msg.LsfRevision != model.LsfRevision`? It looks like now it won't be updated.
Should we either return an error in this case, or apply the new `LsfRevision` for the same `LsfUrl`?


Another case: only `LsfRevision` is set (so that msg.LsfUrl == "").